src/overlays/actors/ovl_Dm_Char09/z_dm_char09.c

```diff
@@ -32,28 +30,197 @@ const ActorInit Dm_Char09_InitVars = {
     (ActorFunc)DmChar09_Draw,
 };
 
-#endif
-
-extern UNK_TYPE D_0600005C;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char09/func_80AB1E10.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char09/DmChar09_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char09/DmChar09_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char09/func_80AB1FA0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char09/func_80AB1FDC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char09/func_80AB2258.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char09/func_80AB2268.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char09/func_80AB24BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char09/DmChar09_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char09/func_80AB25D8.s")
+static AnimationInfo sAnimations[] = {
+    { &object_bee_Anim_00005C, 1.0f, 0.0f, -1.0f, 0, 0.0f },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char09/DmChar09_Draw.s")
+void DmChar09_ChangeAnimation(SkelAnime* skelAnime, AnimationInfo* animation, u16 index) {
+    f32 frameCount;
+
+    animation += index;
+
+    if (animation->frameCount < 0.0f) {
+        frameCount = Animation_GetLastFrame(animation->animation);
+    } else {
+        frameCount = animation->frameCount;
+    }
+
+    Animation_Change(skelAnime, animation->animation, animation->playSpeed, animation->startFrame, frameCount,
+                     animation->mode, animation->morphFrames);
+}
+
+void DmChar09_Init(Actor* thisx, PlayState* play) {
+    DmChar09* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 19.0f);
+    SkelAnime_Init(play, &this->skelAnime, &object_bee_Skel_001398, &object_bee_Anim_00005C, this->jointTable,
+                   this->morphTable, OBJECT_BEE_LIMB_MAX);
+    DmChar09_ChangeAnimation(&this->skelAnime, sAnimations, 0);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->unk_228 = Rand_ZeroOne() * 65535.0f;
+    this->unk_22A = Rand_ZeroOne() * 65535.0f;
+    this->actionFunc = DmChar09_DoNothing;
+}
+
+void DmChar09_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_80AB1FA0(DmChar09* this, s32 arg1) {
+    Math_Vec3s_ToVec3f(&this->actor.world.pos, &this->unk_224[arg1]);
+}
+
+void func_80AB1FDC(DmChar09* this, PlayState* play) {
+    Actor* thisx = &this->actor;
```

Comment:
https://decomp.me/scratch/DjC70
Yeah, i didn't find any other way and this was pointed by other members. Please feel free to try if you can find a better match

Reply:
Not required per se, but seems the best match. Otherwise it needs to use an `if(1){}`. Also this pattern seems to be required regularly in Oot3D so seems more real.